src/sys/select.rs

```diff
@@ -30,8 +32,9 @@ impl FdSet {
         unsafe { libc::FD_CLR(fd, &mut self.0) };
     }
 
-    pub fn contains(&mut self, fd: RawFd) -> bool {
-        unsafe { libc::FD_ISSET(fd, &mut self.0) }
+    pub fn contains(&self, fd: RawFd) -> bool {
+        let mut copy = self.0;
```

Comment:
`libc::FD_ISSET` should be taking a `*const` pointer rather than `*mut`.  I just opened a PR to fix that.
https://github.com/rust-lang/libc/pull/1725

Reply:
As an aside, if we're worried about the cost of this copy, then we might want to consider not deriving `Copy` for `FdSet`.

Reply:
You know, that's an excellent point.  On my system, `FdSet` is 128 bytes big.  Users probably shouldn't copy such a time unless they really need to. _However_ removing `Copy` doesn't necessarily prevent the value from ever being `memcpy`'d, and the consensus is that it's best to impl `Copy` regardless of size.  See https://www.reddit.com/r/rust/comments/2xxjda/when_should_my_type_be_copy/ and https://internals.rust-lang.org/t/pre-rfc-copy-like-trait-for-large-types/11852/14 for some discussion.

Reply:
I think that for now we should proceed without waiting for https://github.com/rust-lang/libc/pull/1725 .  But copying the entire structure on each loop of the iterator is woefully inefficient.  I think you should retain the `&mut` reference instead.

Reply:
Done. This sadly means that the iterator cannot implement `Clone`, but I don't see a good way around that.